src/pcm.cpp

```diff
@@ -1,68 +1,99 @@
+#define DR_FLAC_IMPLEMENTATION
+#include <dr_libs/dr_flac.h>
+#define DR_MP3_IMPLEMENTATION
+#include <dr_libs/dr_mp3.h>
 #define DR_WAV_IMPLEMENTATION
 #include <dr_libs/dr_wav.h>
 
 #include <capo/pcm.hpp>
 #include <algorithm>
+#include <cstring>
 #include <fstream>
+#include <iostream>
 #include <optional>
 
 namespace capo {
 namespace {
-class WAV {
```

Comment:
Why has this been removed?

Reply:
No real reason to have it previously; there was no need to keep track of the dr_libs objects, since only reading the files fully was required. I decided to simplify it using functions instead.

Reply:
There's very much reason to have it, lifetime is critically important. It's also needed (features expanded) in streaming, so I'd suggest trying to incorporate a design around it / using it.

Reply:
I'll push the latest changes to that unstable branch so the potential "future" usage is visible.

Reply:
RAII already handles this. Data is stored and the lifetime is handled by PCM. If an explicit deinit is needed it will occur automatically. The data needed right now is fetched on load so there's no need to hold the dr_libs object at any point in time. And streaming will need a completely different interface anyways because data will be supplied progressively, so there is no reason to keep the old approach in my opinion.